### PR TITLE
box2d: add new version 3.0.0

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **box2d/3.0.0**

#### Motivation
box2d/3.0.0 has been completely overhauled.
It is written in C instead of C++.

#### Details
https://github.com/erincatto/box2d/compare/v2.4.2...v3.0.0

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Comment by toge** [comment]
@uilianries 
Only `pr-merge` has been remaining in ci pipeline.
Could you help me to fix this issue?

**Comment by uilianries** [comment]
@toge I just re-triggered now, thank you for pinging! The previous Conan 2.x build passed, but Conan 1.x was aborted internally by the CI, hope it works now.

**Review by danimtb — DISMISSED** [review]

**Review by uilianries — COMMENTED** [review]
Partially reviewed, +3k lines from upstream's diff, is not something easy :sweat_smile: 

I see the project moved to C in this new version: https://c3i.jfrog.io/c3i/misc-v2/logs/pr/24946/5-linux-gcc/box2d/3.0.0//12b14f6c484feae89100caf3d138013fa9682a4d-build.txt

I would suggest removing C++ from the package ID:

```
def configure(self):
    if Version(self.version) >= "3.0.0":
        del self.settings.compiler.libcxx
        del self.settings.compiler.cppstd
```

File: recipes/box2d/all/conanfile.py [hunk 1]
```diff
@@ -39,15 +50,41 @@ def configure(self):
...
+    def requirements(self):
+        if Version(self.version) >= "3.0.0":
+            self.requires("simde/0.8.2")
+
+    def validate(self):
+        if Version(self.version) < "3.0.0":
+            return
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
```
Comment (AbrilRBS) [inline recipes/box2d/all/conanfile.py:62]: Note that once we drop Conan 1 support, Conan 2 added support for `compiler.cstd`, which could be useful for C projects like this, see https://docs.conan.io/2/reference/tools/build.html#cstd

**Review by AbrilRBS — DISMISSED** [review]
Thanks!

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM. Box2D 3.x has several changes, including moving from C++ to C. More information: https://box2d.org/posts/2023/01/starting-box2d-3.0/

File: recipes/box2d/all/conanfile.py [hunk 2]
```diff
@@ -35,19 +47,48 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             self.options.rm_safe("fPIC")
+        if Version(self.version) >= "3.0.0":
+            del self.settings.compiler.libcxx
+            del self.settings.compiler.cppstd
```
Comment (AbrilRBS) [inline recipes/box2d/all/conanfile.py:52]:
```suggestion
            self.settings.rm_safe("compiler.cppstd")
            self.settings.rm_safe("compiler.libcxx")
```

**Review by AbrilRBS — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Comment by uilianries** [comment]
Big thank you to @toge! :smile:

**Pull request merged**